### PR TITLE
perf(dev): add `mise run test` so local runs match CI

### LINKED ISSUE #6377: perf(dev): add `mise run test` so local runs match CI
Developers need a local full-suite command that matches Mobile CI.

Today `CONTRIBUTING.md` points people at plain `flutter test` for the full app suite. That command compiles and runs a fresh isolate per test file, while Mobile CI uses `very_good test --optimization --concurrency=4 --exclude-tags integration --test-randomize-ordering-seed random` after `flutter pub get` and a pinned `very_good_cli 1.3.0` install.

Acceptance criteria:

- A `mise run test` task exists under `mobile/` for the full app test suite.
- The task runs `flutter pub get`, enforces `very_good_cli 1.3.0`, and invokes the same optimized test flags used by Mobile CI.
- Local docs distinguish single-file iteration with `flutter test path/to/foo_test.dart` from full-suite verification with `mise run test`.
- Optimizer-generated temporary files cannot accidentally land in git.

### PR DESCRIPTION
**Pull request by rabble**
Closes #6377

## Why

`CONTRIBUTING.md` tells developers to run `flutter test` for the suite. That compiles and spins a **fresh isolate per test file** across 1,226 files. CI runs `very_good test --optimization`, which bundles the untagged files into one isolate. Nothing pointed developers at the faster command, and nothing made "green locally" mean the same thing as "green in CI".

## Measured

Head-to-head on an identical subset — 31 files / 277 tests, selected round-robin so it spans directories, same machine, back to back:

| Command | Wall | Tests |
|---|---:|---:|
| `flutter test --concurrency=4` | **60s** | 277 |
| `very_good test --optimization --concurrency=4` | **17s** | 277 |

**3.5× on identical work** — roughly 1.4s per file of isolate and compile overhead the optimizer amortises away.

Method (uses the shard selector from #6375 to hold the file set constant):

```bash
cd mobile
bash scripts/ci/select_test_shard.sh --total 40 --index 0 --force
mise exec -- flutter test --concurrency=4 --exclude-tags integration     # 60s
mise exec -- very_good test --optimization --concurrency=4 --exclude-tags integration  # 17s
git checkout -- test
```

## What changed

- **`mobile/mise.toml`** — new `mise run test` task wrapping the command Mobile CI runs. It runs `flutter pub get`, enforces the pinned `very_good_cli 1.3.0` even when another `very_good` is already on `PATH`, and then invokes the optimized suite with CI flags.
- **`mobile/.gitignore`** — ignores `test/.test_optimizer.dart`, the temporary optimizer bundle file, so a crashed or hard-killed run cannot leave an accidental staged artifact.
- **`CONTRIBUTING.md`** — both the day-to-day and testing-expectations blocks now distinguish `flutter test test/path/to/foo_test.dart` (while iterating on one file) from `mise run test` (whole suite, before pushing), with the measured numbers.
- **`mobile/docs/TEST_PERFORMANCE.md`** — the comparison and its method, next to the existing merged-isolate notes. The local instability note now points at #6372 and asks developers to capture the failing seed before root-causing.

The task takes no path argument on purpose: `very_good test` has no subset flag, so it is all-or-nothing. `flutter test test/path/to/foo_test.dart` stays the right tool for a single file, and the docs say so rather than leaving people to discover it.

## Correctness

Docs and a task wrapper. No app code, gates, ratchets, or workflows are touched. The task now mirrors CI setup (`flutter pub get` + pinned Very Good CLI) before running the same optimized test flags as `mobile_ci.yaml`.

## Verification

- `mise run test` in a fresh takeover worktree detected the existing `very_good_cli 1.1.1`, installed `very_good_cli 1.3.0`, ran `flutter pub get`, and launched the optimized randomized suite.
- Full optimized run reached the real suite and completed with one pre-existing temp-directory cleanup failure in `test/providers/personal_event_cache_service_provider_test.dart` (`Directory not empty`). The same test file passed when rerun directly with the same seed: `flutter test test/providers/personal_event_cache_service_provider_test.dart --test-randomize-ordering-seed=26429946`.
- `git check-ignore -v mobile/test/.test_optimizer.dart` confirms the optimizer artifact is ignored.
- `git diff --check`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by NotThatKindOfDrLiz** [comment]
@rabble I took over the review fixes and rebuilt the branch from current `origin/main` so it no longer conflicts.

Changes pushed in `b976359bf2`:

- `mise run test` now runs `flutter pub get`, enforces `very_good_cli 1.3.0` even if another `very_good` is already on PATH, and uses POSIX-safe `set -eu`.
- `mobile/.gitignore` now ignores `test/.test_optimizer.dart`.
- CONTRIBUTING examples use a copy-pasteable test path instead of shell-redirection syntax.
- The performance note now points at #6372 and asks people to capture the failing seed before root-causing merged-isolate failures.
- #6377 now has acceptance criteria instead of the auto-created stub.

Verification:

- Local `mise run test` exercised the wrapper: it detected `very_good_cli 1.1.1`, installed `1.3.0`, ran `flutter pub get`, and launched the optimized suite. That full local run hit one pre-existing temp-directory cleanup failure in `test/providers/personal_event_cache_service_provider_test.dart`; rerunning that file directly with the same seed passed.
- `git check-ignore -v mobile/test/.test_optimizer.dart`
- `git diff --check`
- GitHub checks are green on `b976359bf2`: mergeability, semantic PR, preview build, generated files, format, analyze, tests, VGV tag gate, and Mobile CI.

Please review the takeover changes and merge when you are ready.

**Review by NotThatKindOfDrLiz — APPROVED** [review]
Approved after takeover. The branch is rebuilt onto current `origin/main`, the local test task now enforces the CI-pinned Very Good CLI version and runs `flutter pub get`, the optimizer artifact is ignored, docs are copy-pasteable, and CI is green on `b976359bf2`.

File: mobile/docs/TEST_PERFORMANCE.md [hunk 1]
```diff
@@ -4,6 +4,31 @@ The app test suite is large enough that fixed startup costs and hidden waits
...
+|---|---:|---:|
+| `flutter test --concurrency=4` | 60s | 277 |
+| `very_good test --optimization --concurrency=4` | 17s | 277 |
+
+About 1.4s per file of isolate and compile overhead that the optimizer
+amortises away. Keep using `flutter test <path>` while iterating on one
+file — the optimizer has no subset flag, so it is all-or-nothing.
+
+Note the optimized full suite can expose pre-existing seed- and load-sensitive
+failures on a developer machine; #6372 tracks that cleanup. If a red run is not
```
Comment (realmeylisdev) [inline mobile/docs/TEST_PERFORMANCE.md:28]: `#6372` is an open PR, and the part of it that documents this instability — `docs/PERF_BASELINE.md`, section "Full-suite run-to-run stability (local)" — only exists on that branch. So this pointer resolves to unmerged work today, and to closed work once it lands.

The cleanup itself is tracked by issues, and #6372's own baseline doc defers to one of them:

- #6280 — `personalEventCacheServiceProvider` under the VGV merged isolate (cited verbatim in `PERF_BASELINE.md` as "a known CI-only flake, #6280")
- #6240 — same service, still order-dependent after #5738/#5837

Citing #6280 / #6240 here keeps the pointer alive regardless of what happens to #6372.

File: mobile/docs/TEST_PERFORMANCE.md [hunk 2]
```diff
@@ -4,6 +4,31 @@ The app test suite is large enough that fixed startup costs and hidden waits
 matter. Use `scripts/test_timing_baseline.sh` before and after test-speed
 changes so improvements are measured instead of guessed.
 
+## Running the suite locally
+
+Use `mise run test` from `mobile/` for a full local run. It wraps the same
+`very_good test --optimization --concurrency=4` command Mobile CI uses, so a
+local pass means the same thing a CI pass does.
```
Comment (realmeylisdev) [inline mobile/docs/TEST_PERFORMANCE.md:11]: "so a local pass means the same thing a CI pass does" is stronger than the rest of this section supports. Line 27 warns the same run surfaces failures unrelated to your diff, and #6372's baseline doc goes further — "Treat a single red local full-suite run as inconclusive and re-run before attributing it to a diff." The seed is `random` on both sides, so the two runs exercise different orderings by construction, and #6280 is a documented case of red-on-CI / green-locally at the same seed.

Something like "reproduces CI's invocation — same optimizer, same flags, same merged-isolate bundling" states what is actually true without promising equivalence.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -232,9 +232,17 @@ blanks yourself and hope review will sort it out later.
 cd mobile
 flutter pub get
 flutter analyze
-flutter test
+flutter test test/path/to/foo_test.dart     # one file or directory
+mise run test           # whole suite, the way CI runs it
```
Comment (realmeylisdev) [inline CONTRIBUTING.md:236]: `README.md:67-69` carries the same three-command block (`flutter pub get` / `flutter analyze` / `flutter test`) and is untouched by this PR. On `main` those two blocks were identical, so after this change the repo's two top-level docs disagree about the exact command this PR is about — and README is the one a new contributor reads first (it already advertises `mise run` tasks eleven lines below). `docs/RELEASE_CHECKLIST.md:24-26` is a third copy of the same block, in the pre-release gate.

`grep -rn "mise run test" --include="*.md" .` returns only `CONTRIBUTING.md` and `mobile/docs/TEST_PERFORMANCE.md`. Mirroring these two lines into those files finishes the edit.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -232,9 +232,17 @@ blanks yourself and hope review will sort it out later.
...
 ```
 
+Use `flutter test test/path/to/foo_test.dart` while iterating on a specific
+file, and `mise run test` for the full suite. `mise run test` wraps `very_good test
+--optimization`, which is what Mobile CI runs: it bundles the untagged test
+files into a single isolate instead of compiling and spinning one per file.
+Measured on an identical 31-file / 277-test subset, same machine, back to
+back: **60s for `flutter test`, 17s for the optimized run**.
```
Comment (realmeylisdev) [inline CONTRIBUTING.md:244]: The 60s/17s pair now lives in three places in this diff — here, `mobile/docs/TEST_PERFORMANCE.md:18-21`, and `mobile/mise.toml:96-98` — and a fourth in #6372's `docs/PERF_BASELINE.md`, which reports the same 3.5x / ~1.4s-per-file from its own run. Since both PRs are in flight together, worth collapsing to one home (with the others linking to it), otherwise the four copies drift independently the first time anyone re-measures.

**Review by realmeylisdev — COMMENTED** [review]
Second pass over the takeover commit. The task is flag-for-flag identical to the CI Tests step (pins, flags, order, working dir), and the wrapper logic holds up under the edge cases I tried — details in the thread replies.

Three doc points inline: one tracker reference that points at unmerged work, one half-finished doc edit, and a wording/duplication nit. None of them block the merge.
